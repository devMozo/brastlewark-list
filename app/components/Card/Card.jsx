import React from 'react';
import { Link } from 'react-router-dom';
const Warrior = require('assets/warrior.png');

export const CardDescription = ({ link = '', title = '', children }) => {
  return (
    <section className="CardDescription">
      <h2> {title} </h2>
      <p className="padding-2-x padding-1_5-bottom"> {children} </p>
      <Link className="padding-1-all font-medium" to={link}>
        Go to the village!
      </Link>
    </section>
  );
};

export const Card = ({ withWhiteBackground, withShadow }) => {
  return (
    <article
      className={`Card ${withWhiteBackground && 'card--white-background'} ${withShadow &&
        'card--shadow'}`}
    >
      <header>
        <img src={String(Warrior)} />
      </header>
      <CardDescription link={'/village/brastlewark'} title="Brastlewark">
        Brastlewark is an almost exclusively gnomish city in eastern Cheliax, located on the western
        edge of the Aspodell Mountains, at the head of the Brastle River. It is believed to be the
        largest gnome settlement in the Inner Sea region, perhaps even in the entire world.
      </CardDescription>
    </article>
  );
};
