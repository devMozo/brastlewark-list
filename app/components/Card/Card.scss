.Card {
  &.card--white-background {
    background-color: white;
  }

  &.card--shadow {
    box-shadow: -1px 15px 30px -12px black;
  }

  header {
    background-image: url('assets/background.jpg');
    img {
      position: relative;
      left: -40px;
      margin-top: -40px;
    }
  }

  .CardDescription {
    text-align: center;
    color: #aaa;

    h2 {
      color: #222;
    }

    a {
      width: 100%;
      background-color: #222;
      color: white;
      display: inline-block;
      box-sizing: border-box;
    }
  }
}
