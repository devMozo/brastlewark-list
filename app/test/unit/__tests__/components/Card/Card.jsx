import React from 'react';
import { Card } from 'components/Card/Card';
import { BrowserRouter } from 'react-router-dom';
import { render } from 'enzyme';

describe('<Card />', () => {
  let props;
  const MOCK_LINK = '/link-test';
  const card = () => {
    return render(
      <BrowserRouter>
        <Card {...props} />
      </BrowserRouter>,
    );
  };

  beforeEach(() => {
    props = {
      withWhiteBackground: undefined,
      withShadow: undefined,
      link: undefined,
    };
  });

  it('Snapshotesting completed <Card />', () => {
    props = {
      withWhiteBackground: true,
      withShadow: true,
      link: MOCK_LINK,
    };
    expect(card()).toMatchSnapshot();
  });

  it('Snapshotesting <Card /> without props', () => {
    expect(card()).toMatchSnapshot();
  });

  it('Check if the link is loaded correctly', () => {
    props.link = MOCK_LINK;
    const aside = card().find(`a[href="${MOCK_LINK}"]`);

    expect(aside.length).toBe(1);
  });

  it('Has always an image', () => {
    const img = card().find('img');

    expect(img.length).toBe(1);
  });
});
