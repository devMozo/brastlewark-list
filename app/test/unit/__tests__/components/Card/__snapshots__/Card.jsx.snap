// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Snapshotesting <Card /> without props 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "Card undefined undefined",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "src": "test-file-stub",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "src": undefined,
            },
          },
        ],
        "name": "header",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "CardDescription",
          },
          "children": Array [
            Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": " Brastlewark ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h2",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "padding-2-x padding-1_5-bottom",
                },
                "children": Array [
                  Object {
                    "data": " Brastlewark is an almost exclusively gnomish city in eastern Cheliax, located on the western edge of the Aspodell Mountains, at the head of the Brastle River. It is believed to be the largest gnome settlement in the Inner Sea region, perhaps even in the entire world. ",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "padding-1-all font-medium",
                    "href": "/",
                  },
                  "children": Array [
                    Object {
                      "data": "Go to the village!",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Object {
              "attribs": Object {
                "class": "padding-2-x padding-1_5-bottom",
              },
              "children": Array [
                Object {
                  "data": " Brastlewark is an almost exclusively gnomish city in eastern Cheliax, located on the western edge of the Aspodell Mountains, at the head of the Brastle River. It is believed to be the largest gnome settlement in the Inner Sea region, perhaps even in the entire world. ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "padding-1-all font-medium",
                  "href": "/",
                },
                "children": Array [
                  Object {
                    "data": "Go to the village!",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "href": undefined,
                },
              },
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": " Brastlewark ",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "h2",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "padding-1-all font-medium",
                "href": "/",
              },
              "children": Array [
                Object {
                  "data": "Go to the village!",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "padding-2-x padding-1_5-bottom",
                },
                "children": Array [
                  Object {
                    "data": " Brastlewark is an almost exclusively gnomish city in eastern Cheliax, located on the western edge of the Aspodell Mountains, at the head of the Brastle River. It is believed to be the largest gnome settlement in the Inner Sea region, perhaps even in the entire world. ",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": " Brastlewark ",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "h2",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "href": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "href": undefined,
              },
            },
          ],
          "name": "section",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {
          "class": "CardDescription",
        },
        "children": Array [
          Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": " Brastlewark ",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h2",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "padding-2-x padding-1_5-bottom",
              },
              "children": Array [
                Object {
                  "data": " Brastlewark is an almost exclusively gnomish city in eastern Cheliax, located on the western edge of the Aspodell Mountains, at the head of the Brastle River. It is believed to be the largest gnome settlement in the Inner Sea region, perhaps even in the entire world. ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "padding-1-all font-medium",
                  "href": "/",
                },
                "children": Array [
                  Object {
                    "data": "Go to the village!",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "href": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          Object {
            "attribs": Object {
              "class": "padding-2-x padding-1_5-bottom",
            },
            "children": Array [
              Object {
                "data": " Brastlewark is an almost exclusively gnomish city in eastern Cheliax, located on the western edge of the Aspodell Mountains, at the head of the Brastle River. It is believed to be the largest gnome settlement in the Inner Sea region, perhaps even in the entire world. ",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "padding-1-all font-medium",
                "href": "/",
              },
              "children": Array [
                Object {
                  "data": "Go to the village!",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "href": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "href": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": " Brastlewark ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h2",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "padding-1-all font-medium",
              "href": "/",
            },
            "children": Array [
              Object {
                "data": "Go to the village!",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "padding-2-x padding-1_5-bottom",
              },
              "children": Array [
                Object {
                  "data": " Brastlewark is an almost exclusively gnomish city in eastern Cheliax, located on the western edge of the Aspodell Mountains, at the head of the Brastle River. It is believed to be the largest gnome settlement in the Inner Sea region, perhaps even in the entire world. ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": " Brastlewark ",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "h2",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "href": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "href": undefined,
            },
          },
        ],
        "name": "section",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "attribs": Object {
                "src": "test-file-stub",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "src": undefined,
              },
              "x-attribsPrefix": Object {
                "src": undefined,
              },
            },
          ],
          "name": "header",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "article",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`<Card /> Snapshotesting completed <Card /> 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "Card card--white-background card--shadow",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "src": "test-file-stub",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "src": undefined,
            },
          },
        ],
        "name": "header",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "CardDescription",
          },
          "children": Array [
            Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": " Brastlewark ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h2",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "padding-2-x padding-1_5-bottom",
                },
                "children": Array [
                  Object {
                    "data": " Brastlewark is an almost exclusively gnomish city in eastern Cheliax, located on the western edge of the Aspodell Mountains, at the head of the Brastle River. It is believed to be the largest gnome settlement in the Inner Sea region, perhaps even in the entire world. ",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "padding-1-all font-medium",
                    "href": "/link-test",
                  },
                  "children": Array [
                    Object {
                      "data": "Go to the village!",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Object {
              "attribs": Object {
                "class": "padding-2-x padding-1_5-bottom",
              },
              "children": Array [
                Object {
                  "data": " Brastlewark is an almost exclusively gnomish city in eastern Cheliax, located on the western edge of the Aspodell Mountains, at the head of the Brastle River. It is believed to be the largest gnome settlement in the Inner Sea region, perhaps even in the entire world. ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "padding-1-all font-medium",
                  "href": "/link-test",
                },
                "children": Array [
                  Object {
                    "data": "Go to the village!",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "href": undefined,
                },
              },
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": " Brastlewark ",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "h2",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "padding-1-all font-medium",
                "href": "/link-test",
              },
              "children": Array [
                Object {
                  "data": "Go to the village!",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "padding-2-x padding-1_5-bottom",
                },
                "children": Array [
                  Object {
                    "data": " Brastlewark is an almost exclusively gnomish city in eastern Cheliax, located on the western edge of the Aspodell Mountains, at the head of the Brastle River. It is believed to be the largest gnome settlement in the Inner Sea region, perhaps even in the entire world. ",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": " Brastlewark ",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "h2",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "href": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "href": undefined,
              },
            },
          ],
          "name": "section",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {
          "class": "CardDescription",
        },
        "children": Array [
          Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": " Brastlewark ",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h2",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "padding-2-x padding-1_5-bottom",
              },
              "children": Array [
                Object {
                  "data": " Brastlewark is an almost exclusively gnomish city in eastern Cheliax, located on the western edge of the Aspodell Mountains, at the head of the Brastle River. It is believed to be the largest gnome settlement in the Inner Sea region, perhaps even in the entire world. ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "padding-1-all font-medium",
                  "href": "/link-test",
                },
                "children": Array [
                  Object {
                    "data": "Go to the village!",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "href": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          Object {
            "attribs": Object {
              "class": "padding-2-x padding-1_5-bottom",
            },
            "children": Array [
              Object {
                "data": " Brastlewark is an almost exclusively gnomish city in eastern Cheliax, located on the western edge of the Aspodell Mountains, at the head of the Brastle River. It is believed to be the largest gnome settlement in the Inner Sea region, perhaps even in the entire world. ",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "padding-1-all font-medium",
                "href": "/link-test",
              },
              "children": Array [
                Object {
                  "data": "Go to the village!",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "a",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "href": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "href": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": " Brastlewark ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h2",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "padding-1-all font-medium",
              "href": "/link-test",
            },
            "children": Array [
              Object {
                "data": "Go to the village!",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "a",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "padding-2-x padding-1_5-bottom",
              },
              "children": Array [
                Object {
                  "data": " Brastlewark is an almost exclusively gnomish city in eastern Cheliax, located on the western edge of the Aspodell Mountains, at the head of the Brastle River. It is believed to be the largest gnome settlement in the Inner Sea region, perhaps even in the entire world. ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": " Brastlewark ",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "h2",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "href": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "href": undefined,
            },
          },
        ],
        "name": "section",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "attribs": Object {
                "src": "test-file-stub",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "src": undefined,
              },
              "x-attribsPrefix": Object {
                "src": undefined,
              },
            },
          ],
          "name": "header",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "article",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
